cmd/tailscale/cli/configure-kube.go

```diff
@@ -44,6 +44,10 @@ See: https://tailscale.com/s/k8s-auth-proxy
 
 // kubeconfigPath returns the path to the kubeconfig file for the current user.
 func kubeconfigPath() string {
+	if kubeconfig := os.Getenv("KUBECONFIG"); kubeconfig != "" {
+		return filepath.SplitList(kubeconfig)[0]
```

Comment:
Thank you very much for the PR @CosmicToast .

I think we should not diverge from the behaviour of kubectl and other tools and libraries in the ecosystem, where if KUBECONFIG is set:
- the config is written to the first existant file in the list if any found
- if none of the files exist, the config is written to the last file in the list

See i.e [kubectl](https://github.com/kubernetes/kubectl/blob/release-1.30/pkg/cmd/config/config.go#L49), [kind](https://github.com/kubernetes-sigs/kind/issues/1060)

I agree that in most use cases only one will be set, but there are also cases when someone/some tool will set it to a list, see i.e https://github.com/helm/helm/issues/11579

Here is the logic that kind uses to parse the list, we could probably do something similar https://github.com/kubernetes-sigs/kind/blob/v0.23.0-alpha/pkg/cluster/internal/kubeconfig/internal/kubeconfig/paths.go

Reply:
Sounds good, I'll get it patched up today or tomorrow.
I have to admit I'm not that familiar with the k8s ecosystem, I'm patching this in for a coworker :)
I'll re-request a review once it's done.

Reply:
Thank you, this is much appreciated!